### PR TITLE
Makes the pet collar not an accessory. Remake remake of the revert of the revert.

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

Makes the pet collar not an accessory so it cannot be put on jumpsuits, and can only be put on pets. This is done in the style of https://github.com/ParadiseSS13/Paradise/pull/15170 as opposed to https://github.com/ParadiseSS13/Paradise/pull/15155 for technical reasons. There was an original PR about this (https://github.com/ParadiseSS13/Paradise/pull/19606) but I managed to muck up my commits something fierce so I'm cheekily remaking it.
## Why It's Good For The Game

As Affected stated in his initial PR, "People only ever add collars to jumpsuits for weird petplay fetish things, which we dont want here. Its time to stop this." I agree with this point,  and Affected's PR was only really reverted in https://github.com/ParadiseSS13/Paradise/pull/15169 for procedure reasons (as far as I can tell at least). In the past, certain items that have been changed/removed for this reason have been defended on the principle that it's entirely a player-problem and that the item is fine to leave, such as with cat ears and the like. For pet collars specifically, I disagree entirely. No sensible employee should be wandering around wearing a collar, and _almost_ every use of this item I've seen has been for either some weird pet-play or something as questionable.
## Testing

Spawned in, bought a pet collar, confirmed it had the correct sprite, made sure IDs could be clipped on to it, and that only one ID could be clipped on at a time (ran into a humorous bug when testing where it would just eat IDs endlessly no matter how many had been clipped on before, effectively deleting them). Slapped the collar on Ian and ensured it both appeared and allowed the lovely Corgi to open doors with its access. Lastly, attempted to put the collar on my jumpsuit both by clicking the jumpsuit and clicking myself. No collar attached.
## Changelog
:cl:
del: You can no longer put pet collars on jumpsuits
/:cl:
Thank you to Farie, Lewcc, and Contrabang for their help on https://github.com/ParadiseSS13/Paradise/pull/19606. I apologize about having to remake the PR, but I seemed to have a very big misunderstanding about how to update my code/deconflict and stuff.

### CODE REVIEW

File: code/game/machinery/vending.dm
Comment (Sirryan2002) [inline code/game/machinery/vending.dm:1881]:
```suggestion
	products = list(/obj/item/petcollar = 5, /obj/item/storage/firstaid/aquatic_kit/full = 5, /obj/item/fish_eggs/goldfish = 5,
```

**Comment by Pixelzilla** [comment]
> "People only ever add collars to jumpsuits for weird petplay fetish things, which we dont want here. Its time to stop this."

I have yet to see actual instances of this happening and if it is indeed happening it is an issue with specific players that should be delt with on an individual basis as it is a clear server rulebreak and not an issue with the item itself. This idea of removing items/features that in themselves are perfectly fine just because a tiny subset of people have or could abuse it is silly. 

Its also worth pointing out that making blanket statements like this and saying everyone who uses this feature of a pet collar is acting out some sort of fetish is pretty narrow minded and seems exceptionally biased. I'd like to see the feature retained and appropriate administrative action taken against people who are actually abusing it. I suspect, however, that this isn't quite as big of an issue as it is being made into.

**Comment by SpringSkipper** [comment]
> In the past, certain items that have been changed/removed for this reason have been defended on the principle that it's entirely a player-problem and that the item is fine to leave, such as with cat ears and the like.

Yeah this is my objection to the PR, and it really doesn't feel like you've adequately dispelled that reasoning. I've never seen people wear it for 'weird petplay fetish things'. I think you're just painting with a broad brush here based on something you've heard or saw one person do.

Wearing a collar is weird, but it's something people do in real life especially with goth or punk dress styles. It's not any more weird than bunny ears, a chicken suit, or animal masks.

**Review by lewcc — APPROVED** [review]
Looks good to me

**Comment by LynxSolstice** [comment]
Be cool if these were replaced with lanyards, ID storage on an accessory is useful

**Comment by farie82** [comment]
Putting the PR on hold since it touches vendors. The on hold label will be removed once the economy TM is done/merged

**Comment by Miraviel** [comment]
Can't you put an ID in the collar and then have it on you, de facto having a near-invisible source of access?

The only way to tell if you are wearing an ID this way is if someone examines you and looks at your jumpsuit accessory. Even then I think the collar does not list the details of the ID (or the ID, at all)...

If taj/vulp players want to have a collar accessory, I think they should not be using the pet collar but a different item, preferably one they can recolour.

File: code/game/objects/items/misc.dm
Comment (hal9000PR) [inline code/game/objects/items/misc.dm:40]:
```suggestion
/obj/item/petcollar/attack_self(mob/user)
```

**Pull request merged**
